rspace/src/test/scala/coop/rchain/rspace/test/ArbitraryInstances.scala

```diff
@@ -92,13 +115,42 @@ object ArbitraryInstances {
           {
             val constrainedSize = if (size > 1) size else 1
             for {
-              chans   <- Gen.containerOfN[List, String](constrainedSize, Arbitrary.arbitrary[String])
-              pats    <- Gen.containerOfN[List, Pattern](constrainedSize, Arbitrary.arbitrary[Pattern])
-              boolean <- Arbitrary.arbitrary[Boolean]
-            } yield (chans, WaitingContinuation(pats, new StringsCaptor, boolean))
+              chans <- Gen.containerOfN[List, String](constrainedSize, Arbitrary.arbitrary[String])
+              wc    <- Arbitrary.arbitrary[WaitingContinuation[Pattern, StringsCaptor]]
```

Comment:
I think I'd prefer to take a defensive approach here to ensure that the `List[String]` (channels) is always the same length as the `List[Pattern]` (patterns):
```scala
  implicit val arbitraryNonEmptyMapListStringWaitingContinuation
    : Arbitrary[Map[List[String], WaitingContinuation[Pattern, StringsCaptor]]] = {
    Arbitrary(Gen
      .sized { _ =>
        Gen.nonEmptyContainerOf[List, (List[String], WaitingContinuation[Pattern, StringsCaptor])](
          for {
            wc    <- Arbitrary.arbitrary[WaitingContinuation[Pattern, StringsCaptor]]
            chans <- Gen.containerOfN[List, String](wc.patterns.length, Arbitrary.arbitrary[String])
          } yield (chans, wc)
        )
      }
      .map(_.toMap))
  }
```

Reply:
Done